packages/flutter/test/material/dialog_test.dart

```diff
@@ -102,6 +102,53 @@ void main() {
     expect(materialWidget.type, MaterialType.card);
     expect(materialWidget.elevation, 24);
     expect(materialWidget.color, Colors.grey[800]);
+    expect(materialWidget.shape, const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(2.0))));
+  });
+
+  testWidgets('Custom dialog shape', (WidgetTester tester) async {
+    const RoundedRectangleBorder customBorder =
+      RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0)));
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(brightness: Brightness.dark),
+        home: Material(
+          child: Builder(
+            builder: (BuildContext context) {
+              return Center(
+                child: RaisedButton(
+                  child: const Text('X'),
+                  onPressed: () {
+                    showDialog<void>(
+                      context: context,
+                      builder: (BuildContext context) {
+                        return const AlertDialog(
+                          title: Text('Title'),
+                          content: Text('Y'),
+                          actions: <Widget>[ ],
+                          shape: customBorder,
+                        );
+                      },
+                    );
+                  },
+                ),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('X'));
+    await tester.pump(); // start animation
+    await tester.pump(const Duration(seconds: 1));
+
+    final StatefulElement widget = tester.element(find.byType(Material).last);
```

Comment:
Instead of relying on the Material you're looking for being the last one, you could look for the one that was a descendant (https://docs.flutter.io/flutter/flutter_test/CommonFinders/descendant.html) of the AlertDialog.

Reply:
Great idea, applied to test above as well.